Light (Preview) Font color change for text in tabs

mvm-sap opened the pull request:
Font color of all tabs selected or unselected are set to black  #000000 with this change to have a better contrast.

Before :
![image](https://github.com/user-attachments/assets/f919a906-17cc-4453-8357-35752d446cbf)

After:
![image](https://github.com/user-attachments/assets/f835c75e-4c25-4e57-a981-0a5871095da7)

mvm-sap committed:
Light (Preview) Font color changes in tabs

Font color of all tabs selected or unselected are set to black  #000000
with this change

BeckerWdf commented:
how is this different to https://github.com/eclipse-platform/eclipse.platform.ui/pull/2187 ?

mvm-sap commented:
> how is this different to #2187 ?

In #2187  
Unselected Tabs in view and editor had  #3b3b3b
and 
Selected Tabs in view and editor had #000000

But with this PR 
All tabs, selected and unselected both in editor and view has #000000

We need to delete #2187

BeckerWdf commented:
looks good on macOS for me.
@jukzi: Can you pls. also test and say if this addresses your issues?

jukzi commented:
looks better, but it's hard to see which secondary tab is active:
![image](https://github.com/user-attachments/assets/9461ccb6-3722-4e2d-a60b-f7f1ff80cffd)
used to look like this where it was obvious which tab is the active view by background color:
![image](https://github.com/user-attachments/assets/4aaf2ad2-3a05-4e15-80ed-e8aa0259beb6)

BeckerWdf commented:
> but it's hard to see which secondary tab is active

Do you mean the "Console" tab in your screenshot? It's now indicated by the grey underline.

jukzi commented:
> Do you mean the "Console" tab in your screenshot? It's now indicated by the grey underline.

yes. Unfortunately that grey underline is not as prominent as the changed background. isn't there even some option to diable that underline? Then it would be totally unclear. 
Maybe i can get used to the underline, but at a first glance the old style was more expressive.

BeckerWdf commented:
> > Do you mean the "Console" tab in your screenshot? It's now indicated by the grey underline.
> 
> yes. Unfortunately that grey underline is not as prominent as the changed background. isn't there even some option to diable that underline? Then it would be totally unclear. Maybe i can get used to the underline, but at a first glance the old style was more expressive.

Yes that's true but that's the intention of that change. As it's not the "primary active tab" but only the "secondary active tab" is should be not as prominent. We want to give the "main actor" more focus and the "secondary active tab" is not only a "supporting actor". But we could make the grey underline a bit darker grey to make it a bit more prominent but not _too_ prominent.
What about something like this:
<img width="873" alt="Screenshot 2024-09-19 at 09 39 27" src="https://github.com/user-attachments/assets/a25e380b-43ab-4d75-bf65-efa9462b2efd">
compared to the current state:
<img width="873" alt="Screenshot 2024-09-19 at 09 40 53" src="https://github.com/user-attachments/assets/caf8eaef-ac3f-4555-929a-eda2edd1d12f">

jukzi commented:
darker looks better. maybe even one pixel thicker?

jukzi commented:
The best would be if the high contrast icons of the inactive  tabs gets reduced :-) just as idea

BeckerWdf commented:
> The best would be if the high contrast icons of the inactive tabs gets reduced

That's a nice idea but currently not possible to implement. We could save this for a further improvement via a separate PR.

BeckerWdf commented:
> maybe even one pixel thicker?

We just made the line thinner in the last release. This is something we can improve (and maybe make it bigger again) later on.

BeckerWdf committed:
Make Grey Highlight of Inactive Tabs Darker

To make it easier to see which one is the "secondary active tab"
(the active tab in inactive part stacks) the grey line in that tab
is made a bit darker.

BeckerWdf commented:
> darker looks better. maybe even one pixel thicker?

I just pushed a second commit into this pr making the grey darker.
Now looks like the one on the right side. Left side is the current grey color:
<img width="1301" alt="image" src="https://github.com/user-attachments/assets/f3debabc-e2a1-4ee5-ad84-d967ea364324">

BeckerWdf commented:
https://ci.eclipse.org/platform/job/eclipse.platform.ui/job/PR-2294/4/console reports a lot of API breakages.
They are not caused by this PR and for it looks like false alarms.
Does anybody have a clue what's happening?

jukzi commented:
> Does anybody have a clue what's happening?

may be a random fail retry.

jukzi reviewed (approved):
LGTM
